Algorithm.Framework/Portfolio/MaximumSharpeRatioPortfolioOptimizer.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using Accord.Math;
+using Accord.Math.Optimization;
+using Accord.Statistics;
+
+namespace QuantConnect.Algorithm.Framework.Portfolio
+{
+    public class MaximumSharpeRatioPortfolioOptimizer : MinimumVariancePortfolioOptimizer
```

Comment:
I think it should be the other way around: `MinimumVariancePortfolioOptimizer` inherits from `MaximumSharpeRatioPortfolioOptimizer`, because the first has more constraints than the later.
We have chosen to not do it in python so that each model could be self-contained.

Reply:
There are the same number of constraints in both solvers, but they differ how one of them defined:
- MVO:  r^T x = R
- MSR:  (r -r_f)^T x = 1

where x - weights, r - expected returns, R - target return, r_f - risk-free rate